internal/flags/flags.go

```diff
@@ -0,0 +1,114 @@
+package flags
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/avointsev/yp7m-go/internal/logger"
+)
+
+type AgentConfig struct {
+	Address        string
+	ReportInterval time.Duration
+	PollInterval   time.Duration
+}
+
+type ServerConfig struct {
+	Address string
+}
+
+func GetEnvOrFlag(envVar string, flagValue string, defaultValue string) string {
+	if value, ok := os.LookupEnv(envVar); ok {
+		return value
+	}
+	if flagValue != "" {
+		return flagValue
+	}
+	return defaultValue
+}
+
+func GetIntEnvOrFlag(envVar string, flagValue int, defaultValue int) int {
+	if value, ok := os.LookupEnv(envVar); ok {
+		if intValue, err := strconv.Atoi(value); err == nil {
```

Comment:
Лучше сделать проверку наоборот
```suggestion
		if intValue, err := strconv.Atoi(value); err != nil {
```